### ISSUE TITLE
Is it possible to use the raw event stream (Neuromorphic dataset) to train an SNN model without using frame reconstruction?

### ISSUE THREAD

**Issue by djessy1998** [fangwei123456/SpikingFlow#404]
**Issue type**

- [ ] Bug Report
- [ ] Feature Request
- [x] Help wanted
- [ ] Other

**SpikingJelly version**

`0.0.0.0.14`

**Description**

Hi! First of all, thanks for your work. I have a question related to training a simple SNN, for example. I'm using a neuromorphic dataset where I would like to train my model using the event stream directly, asynchronously, without frame reconstruction. I don't think I've seen something like this in your examples.

**Question**: Is it possible to do this with SpikingJelly?

Also, for classification on DVS128Gesture, could the results obtained using the raw event stream be radically different from the results with frame reconstruction?

Thank you for your answers!

**Comment by fangwei123456** [comment]
> Is it possible to do this with SpikingJelly?

Yes, you can convert the `(x, y, t, p)` to the tensor format, and send these events to the SNN.

But the number of time steps will be in millions, and it seems impossible to train the SNN. Thus, the downampling on time is necessary.

**Comment by SanderDebets** [comment]
Hi @fangwei123456 

Is the conversion from (x, y, t, p) to tensor format something built into SpikingJelly or would it have to be done manually?
I have the same question regarding the downsampling, can it be done with SpikingJelly or does it have to be done manually?
Thanks!

**Comment by SanderDebets** [comment]
Thanks @fangwei123456, I now used torch.stack to create a PyTorch tensor from the dictionary. I don't fully understand what you mean by "if you want to down frames converted from events". Do you mean the frames obtained when setting data_type='frame', or do you mean something else? Indeed I want to downsample the temporal resolution of the events, such that I can then feed it into an SNN. Would it be possible to train an SNN with spikingjelly in this way and obtain comparable accuracy to the frame based example at https://spikingjelly.readthedocs.io/zh-cn/latest/activation_based_en/classify_dvsg.html? My idea would be to downsample to a resolution of 1ms and then classify gestures in time windows of 500ms.
